### PR TITLE
Add authenticator_plugin dev documentation and updates for SAML plugin

### PR DESCRIPTION
**Pull request by john-westcott-iv**

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -75,6 +75,22 @@ SOCIAL_AUTH_LOGIN_REDIRECT_URL = "/api/v1/me"
...
+This setting points to a function which needs to return a dictionary of settings/values like:
+```
+def load_social_auth_settings():
+    logger.info("Loading Gateway social auth settings")
+    return {
+        "SOCIAL_AUTH_USERNAME_IS_FULL_EMAIL": get_preference_value('social_auth', 'SOCIAL_AUTH_USERNAME_IS_FULL_EMAIL', encrypted=False)
+    }
+```
+
+Any settings additional settings supplied by this function will be applied to out default SocialAuth strategy strategy(ansible_base.authentication.social_auth.AuthenticatorStrategy) and will thus be available to the social-core libraries at runtime.
```
Comment (slemrmartin) [inline docs/authentication.md]: "Any  ~settings~ additional settings"

File: docs/authenticator_plugins.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Authenticator Plugins
+
+django-ansible-base provides a plugable authentication system (see authentication.md for high level details) that are backed by individual plugin files. The plugin files are meant to provide access to specific authentication methods like SAML, ldap, OIDC, tacacs, etc. 
+
+This document outlines how to create authenticator plugins.
+
+## Creating an authenticator plugin
+
+### The File
+Authenticator plugins in django-ansible-base should be individual files in the ansible_base/authenticator_plugins folder. That being said, you can configure django-ansible-base to look in additional directories for the authenticator_plugins (see authentication.md). The file should be a logical name for the backend service you are communicating with; for example, saml.pl houses the code for connecting to SAML.
```
Comment (slemrmartin) [inline docs/authenticator_plugins.md]: did you mean `saml.py` ?

**Review by slemrmartin — APPROVED** [review]
Just a nit, otherwise it looks great. 
I learned from this doc :+1:

File: docs/authenticator_plugins.md (hunk 2 above)
Comment (relrod) [reply docs/authenticator_plugins.md]: Oh you didn't hear? We're rewriting the entire project in Perl. 🤣 j/k

**Pull request merged**